account_invoice_date_due/models/account_move.py

```diff
@@ -0,0 +1,24 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    # Show invoice date due even when payment term is defined
+    invoice_date_due_payment_term = fields.Date(
+        related="invoice_date_due", string="Due Date Payment Term"
+    )
+
+    @api.onchange("invoice_date_due")
+    def _onchange_invoice_date_due(self):
+        # Set due date on account move lines
+        if self.invoice_date_due:
+            self.mapped("line_ids").date_maturity = self.invoice_date_due
```

Comment:
In my functional test, I changed the due date of the invoice but the maturity date wasn't set. Any way, shouldn't you set it only for the debit lines?